build/helper/metadata_add_all.py

```diff
@@ -77,13 +73,27 @@ def _add_is_error_handling(f):
 
 def _add_buffer_info(parameter):
     '''Adds buffer information to the parameter metadata iff 'size' is defined else assume not a buffer'''
+
+    # We are going to treat ViChar[], ViString, ViConstString and ViRsrc the same. For simplicity in the generated
+    # code, we will treat them as 'ViChar' and is_buffer true
+    t = parameter['type']
+    if (t.find('[ ]') > 0) or (t.find('[]') > 0) or t == 'ViString' or t == 'ViConstString' or t == 'ViRsrc':
+        parameter['type'] = 'ViChar'
```

Comment:
I wish we weren't modifying metadata, only adding to it. I do see you preserve original info in 'original_type'.

Suggestion: In metadata extraction, instead of type, use something else (c_api_type?), then add type. I can see this being a different PR however.

If you agree, let's open the Issue.

Reply:
I think that is a reasonable change, but it should be it's own PR. It would touch many places just for that.